example/src/AuthenticationMiddleware.ts

```diff
@@ -0,0 +1,12 @@
+import { SlushyAuthenticationMiddleware, SlushyResponse, SlushyRequest } from '@slushy/server';
+
+export class AuthenticationMiddleware implements SlushyAuthenticationMiddleware {
+    execute(req: SlushyRequest, res: SlushyResponse, next: (error?: Error) => any): any {
+        if (req.headers.authorization) {
+            next()
+        } else {
+            res.status(401).json({ error: { message: "unauthorized" }})
```

Comment:
Ideally the library user doesn't want to do this, but instead generate an error e.g.
```ts
new ServerUnauthorizedError({ typedPayload: '' })
```
that contains the possible error codes and response bodies defined in the swagger schema.
Can we showcase this somehow? Maybe we need to extend some of the type generation things for that? 
I think the only response that is not in the swagger schema and still allowed is 500 with an empty response body.

Reply:
![image](https://user-images.githubusercontent.com/2048803/59042652-27b46700-887b-11e9-9088-bc254b70c2a6.png)

Reply:
I think that's the part we actually need to implement for this, so let's do it the next time! :)

Reply:
Yeah. We have to generate those types if declared in OpenAPI file. Will work on that!

Reply:
FYI these types are generated now :)